tests/testAutopas/tests/autopasInterface/Newton3OnOffTest.cpp

```diff
@@ -113,131 +112,92 @@ void Newton3OnOffTest::countFunctorCalls(autopas::ContainerOption containerOptio
-
-      // newton3 variant should not happen
-      EXPECT_CALL(mockFunctor, AoSFunctor(_, _, true)).Times(0);
-      iterate(container,
-              autopas::TraversalSelector<FPCell>::template generateTraversal<MockFunctor<Particle, FPCell>,
-                                                                             autopas::DataLayoutOption::aos, false>(
-                  traversalOption, mockFunctor, traversalSelectorInfo),
-              dataLayout, autopas::Newton3Option::disabled, &mockFunctor);
+          .WillRepeatedly(testing::InvokeWithoutArgs([&]() { callsPair++; }));
+
+      // non useNewton3 variant should not happen
+      EXPECT_CALL(mockFunctor, AoSFunctor(_, _, not useNewton3)).Times(0);
+      iterate(
+          container,
+          autopas::TraversalSelector<FPCell>::template generateTraversal<MockFunctor<Particle, FPCell>,
+                                                                         autopas::DataLayoutOption::aos, useNewton3>(
+              traversalOption, mockFunctor, traversalSelectorInfo),
+          dataLayout, n3option, &mockFunctor);
       break;
     }
     case autopas::DataLayoutOption::cuda: {
       // supresses Warning
       // TODO implement suitable test
-      iterate(container,
-              autopas::TraversalSelector<FPCell>::template generateTraversal<MockFunctor<Particle, FPCell>,
-                                                                             autopas::DataLayoutOption::aos, true>(
-                  traversalOption, mockFunctor, traversalSelectorInfo),
-              dataLayout, autopas::Newton3Option::enabled, &mockFunctor);
+      iterate(
+          container,
+          autopas::TraversalSelector<FPCell>::template generateTraversal<MockFunctor<Particle, FPCell>,
+                                                                         autopas::DataLayoutOption::cuda, useNewton3>(
+              traversalOption, mockFunctor, traversalSelectorInfo),
+          dataLayout, n3option, &mockFunctor);
       break;
     }
     default:
-      FAIL() << "This test does not support data layout : " << autopas::utils::StringUtils::to_string(dataLayout);
-  }
-
-  if (autopas::DataLayoutOption::soa) {
-    // within one cell no N3 optimization
-    EXPECT_EQ(callsNewton3SC, callsNonNewton3SC) << "for containeroption: " << containerOption;
-  }
-  // should be called exactly two times
-  EXPECT_EQ(callsNewton3Pair * 2, callsNonNewton3Pair) << "for containeroption: " << containerOption;
-
-  if (::testing::Test::HasFailure()) {
-    std::cerr << "Failures for Container: " << autopas::utils::StringUtils::to_string(containerOption)
-              << ", Traversal: " << autopas::utils::StringUtils::to_string(traversalOption)
-              << ", Data Layout: " << autopas::utils::StringUtils::to_string(dataLayout) << std::endl;
+      std::cerr << "This test does not support data layout : " << autopas::utils::StringUtils::to_string(dataLayout);
```

Comment:
An invalid test should not pass can you somehow skip this? Fail would be ok for me too if skipping is not possible.

Reply:
I couldn't add a fatal failure (`FAIL()`), since the containing function doesn't return void, but I added a non-fatal failure.